chat-server/src/integrations/openai.ts

```diff
@@ -0,0 +1,41 @@
+// TODO: Consider refactoring to use new Azure OpenAI client SDK - https://www.npmjs.com/package/@azure/openai
+import axios from "axios";
+import { CreateEmbeddingResponse } from "openai";
+interface CreateEmbeddingParams {
+  input: string;
+  user: string;
+}
+export class OpenAIClient {
+  resourcePath: string;
+  apiKey: string;
+  apiVersion: string;
+  constructor(
+    basePath: string,
+    deployment: string,
+    apiKey: string,
+    apiVersion: string
+  ) {
+    this.resourcePath = basePath + deployment;
+    this.apiKey = apiKey;
+    this.apiVersion = apiVersion;
+  }
+
+  embeddings = {
+    create: async ({ input, user }: CreateEmbeddingParams) => {
```

Comment:
I think it would merit declaring an explicit return type here. I would even recommend creating an exported `interface` (or `type`) for like `EmbeddingsService` or whatever this is, then say:
```js
embeddings: EmbeddingsService = { ... }
```

Reply:
will add

Reply:
actually refactoring Embedding service to be a completely separate entity from LLM service. doing b/c while they're logically linked with the OpenAI service, they're less linked in azure, being separate deployments.